app/assets/javascripts/discourse/widgets/widget.js.es6

```diff
@@ -270,18 +196,26 @@ export default class Widget {
     }
   }
 
-  attach(widgetName, attrs, opts) {
-    let WidgetClass = _registry[widgetName];
+  attach(widgetNames, attrs, opts) {
```

Comment:
I'm not sure I agree with this change. 

There are only two places in the commit that adds multiple widgets, and both are only adding two. It would be fewer lines of code to call `this.attach` twice than building an array and calling it this way, not to mention that `this.attach` has to do less code (checking if input is array or not, then looping through it.)

Reply:
This isn't adding multiple widgets, it's adding one widget using the first type that exists. So, for example, since `liked-notification-item` exists, it uses that, whereas there is no `watching-first-post-notification-item`, so the second option, `default-notification-item`, is used instead.

I could still change this back and throw a more specialized error and then, in those two places, do:

```javascript
try {
  this.attach(specialized_name, attrs);
} catch (e) {
  if (e isinstanceof WidgetNotFoundError) {
    this.attach("default-notification-item", attrs);
  } else {
    throw e;
  }
}
```

But I was trying to avoid that.

Reply:
Oh that was not clear to me based on the `widgetNames` parameter! I would expect that to include them all, not the first one it finds.

Instead I'd recommend an option that can be used if the widget is not found. Something like `defaultWidgetName: 'default-notification-item'`